src/main/java/org/team1540/robot2025/subsystems/intake/IntakeIOReal.java

```diff
@@ -0,0 +1,122 @@
+        pivotTalonFXConfigs.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+        pivotTalonFXConfigs.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+
+        Slot0Configs pivotConfigs = pivotTalonFXConfigs.Slot0;
+
+        pivotConfigs.kS = IntakeConstants.PIVOT_INTAKE_KS;
+        pivotConfigs.kV = IntakeConstants.PIVOT_INTAKE_KV;
+        pivotConfigs.kA = IntakeConstants.PIVOT_INTAKE_KA;
+        pivotConfigs.kP = IntakeConstants.PIVOT_INTAKE_KP;
+        pivotConfigs.kI = IntakeConstants.PIVOT_INTAKE_KI;
+        pivotConfigs.kD = IntakeConstants.PIVOT_INTAKE_KD;
+
+        MotionMagicConfigs motionMagicConfigs = pivotTalonFXConfigs.MotionMagic;
+
+        motionMagicConfigs.MotionMagicCruiseVelocity = IntakeConstants.PIVOT_INTAKE_CRUISE_VELOCITY;
+        motionMagicConfigs.MotionMagicAcceleration = IntakeConstants.PIVOT_INTAKE_ACCELERATION;
+
+        pivotFalcon.getConfigurator().apply(pivotConfigs);
+
+        pivotFalcon.setPosition(0);
+
+        funnelNEOConfig.secondaryCurrentLimit(70);
+        funnelNEOConfig.inverted(false);
+        funnelNEOConfig.idleMode(SparkBaseConfig.IdleMode.kCoast);
+
+        funnelNeo.configure(
+                funnelNEOConfig, SparkBase.ResetMode.kResetSafeParameters, SparkBase.PersistMode.kPersistParameters);
+    }
+
+    @Override
+    public void setPivot(Rotation2d rotations) {
+        pivotFalcon.setControl(pivotPositionRequest.withPosition(rotations.getRotations()));
+    }
+
+    @Override
+    public void setRollerVoltage(double voltage) {
+        spinFalcon.setVoltage(voltage);
+    }
+
+    @Override
+    public void setFunnelVoltage(double voltage) {
+        funnelNeo.setVoltage(voltage);
+    }
+
+    @Override
+    public void updateInputs(CoralIntakeInputs inputs) {
+
+        inputs.spinMotorPosition = spinFalcon.getPosition().getValueAsDouble();
+        inputs.spinMotorVelocityRPS = spinFalcon.getVelocity().getValueAsDouble();
+        inputs.spinMotorAppliedVolts = spinFalcon.getMotorVoltage().getValueAsDouble();
+        inputs.spinCurrentAmps = spinFalcon.getStatorCurrent().getValueAsDouble();
+        spinFalconStatus = spinFalcon.getMotorOutputStatus().getStatus();
+
+        inputs.pivotMotorPosition = pivotFalcon.getPosition().getValueAsDouble();
+        inputs.pivotMotorVelocityRPS = pivotFalcon.getVelocity().getValueAsDouble();
+        inputs.pivotMotorAppliedVolts = pivotFalcon.getMotorVoltage().getValueAsDouble();
+        inputs.pivotCurrentAmps = pivotFalcon.getStatorCurrent().getValueAsDouble();
+        pivotFalconStatus = pivotFalcon.getMotorOutputStatus().getStatus();
+
+        inputs.funnelMotorPosition = funnelNeo.getEncoder().getPosition();
+        inputs.funnelMotorVelocityRPS = funnelNeo.getEncoder().getVelocity();
+        inputs.funnelCurrentAmps = funnelNeo.getOutputCurrent();
+        inputs.funnelOutputVoltage = (funnelNeo.getAppliedOutput() * funnelNeo.getBusVoltage());
```

Comment:
This still isn't fixed. For an example of caching status signals, look at [ArmIOTalonFX](https://github.com/flamingchickens1540/robot2025/blob/main/src/main/java/org/team1540/robot2025/subsystems/elevator/ElevatorIOTalonFX.java) or [ElevatorIOTalonFX](https://github.com/flamingchickens1540/robot2025/blob/main/src/main/java/org/team1540/robot2025/subsystems/elevator/ElevatorIOTalonFX.java).

Also, you're continuously calling `getEncoder` on the NEO every loop cycle, which is slow. Call `getEncoder` once on the Neo in the constructor, and store it in a field.